### PR TITLE
feat: create table card component

### PR DESCRIPTION
**Pull request by dominikx96**
# Pull Request Template

## Issue Overview

This PR addresses #2746

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds a table card component: https://www.figma.com/file/Y8bFd0aSz7eYJ2TCNid6s3/bloom-partners?node-id=5875%3A185029

## How Can This Be Tested/Reviewed?
1. Open storybook
2. Compare component with designs

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: ui-components/__tests__/tables/TableCard.test.tsx
Comment (emilyjablonski) [inline ui-components/__tests__/tables/TableCard.test.tsx:7]: Great tests!

File: ui-components/src/tables/TableCard.tsx
Comment (emilyjablonski) [inline ui-components/src/tables/TableCard.tsx:32]: We're moving away from having translated strings in ui-components as any consumer might have a different translation service. An example of this change would be in the `Contact` component where we can pass a strings object as a prop.

File: ui-components/src/tables/TableCard.tsx
Comment (emilyjablonski) [inline ui-components/src/tables/TableCard.tsx:10]: This prop is really tied to our use for this component - that it will add / remove items, but visually this component could be used for other things. Could we toggle that `add items` state on/off through the children? Open to other suggestions!

File: ui-components/src/tables/TableCard.tsx
Comment (emilyjablonski) [inline ui-components/src/tables/TableCard.tsx:14]: As a move toward generalizing our components, I'd suggest a name that doesn't rely on this component having a table as its children. Maybe just something like `Card`?

File: ui-components/src/tables/TableCard.scss
Comment (emilyjablonski) [inline ui-components/src/tables/TableCard.scss:1]: 🥳

File: ui-components/src/locales/general.json
Comment (emilyjablonski) [inline ui-components/src/locales/general.json:834]: Once these strings are moved out of the component, we can also move them into the partners overrides strings file since they won't be used on the public side.

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Screen Shot 2022-06-01 at 10 03 47 AM
I have a couple code style comments, and then it looks like we are missing just a few styles from the Figma designs (the row dividers should reach all the way across, less padding, and ensuring it supports the ability to have that action row that has no header text). Let me know if you have any questions!

File: ui-components/src/tables/TableCard.tsx
Comment (dominikx96) [reply ui-components/src/tables/TableCard.tsx:14]: Make sense @emilyjablonski ! I think we can follow other components and use the `StandardCard` name, what do you think?

File: ui-components/src/tables/TableCard.tsx
Comment (dominikx96) [reply ui-components/src/tables/TableCard.tsx:10]: To make this component more reusable, I can get rid of `Add item` button and just create a prop like `footer: React.ReactNode`. You can pass everything in the parent component (e.g button), but the top padding still is controlled by this card component. What do you think?

**Comment by dominikx96** [comment]
@emilyjablonski thanks for the review! 🚀 I've implemented everything listed above + decided to get rid of the `elementsQty ` prop - we can show/hide an empty state by checking in children exists. I assume that a table content will be controlled by `useState`, so we can render the table conditionally depending on state length 🤔

File: ui-components/src/tables/TableCard.tsx
Comment (emilyjablonski) [reply ui-components/src/tables/TableCard.tsx]: That sounds great!

**Comment by jaredcwhite** [comment]
Hey @dominikx96, awesome to see the updated 2nd gen design tokens getting worked in! I'm wondering if we could add a `.standard-card` class itself to the stylesheet and add a few specific tokens/variables for the component? For example, the border radius and border width/color are currently hardcoded into the stylesheet, rather than coming from component-specific default tokens which could be overridden by a consumer. I guess it depends on which visual aspects we think a consumer might want to customize, but judging from the direction Detroit took with a lot of the components, the answer seems to be "almost everything"… 😅

**Review by emilyjablonski — APPROVED** [review]
This looks great!! Would just want to checkin with Jared about his comments before merge 👍

**Comment by dominikx96** [comment]
> 

Hey @jaredcwhite , make sense! I checked 2nd theming docs again and decided to update component variables + story docs. I love this approach, it's much cleaner!

**Review by jaredcwhite — APPROVED** [review]
Thanks @dominikx96, looks good to me! I wonder if the padding should also get tokenized, but otherwise LGTM. 👍

**Pull request merged**
